lib/core/src/lib/notifications/components/notification-history.component.spec.ts

```diff
@@ -182,4 +184,41 @@ describe('Notification History Component', () => {
             });
         }, 45000);
     });
+
+    it('should return true when there are unread notifications', () => {
+        component.unreadNotifications = [
+            {
+                type: NOTIFICATION_TYPE.INFO,
+                icon: 'info',
+                datetime: new Date(),
+                initiator: { key: '*', displayName: 'SYSTEM' },
+                messages: ['Moved 1 item.'],
+                read: false
+            },
+            {
+                type: NOTIFICATION_TYPE.INFO,
+                icon: 'info',
+                datetime: new Date(),
+                initiator: { key: '*', displayName: 'SYSTEM' },
+                messages: ['Copied 1 item.'],
+                read: false
+            }
+        ];
+
+        const result = component.badgeVisibility();
+        const matIconDebugElement = fixture.debugElement.query(By.css('[matBadge]')).nativeElement;
+
+        expect(result).toBe(true);
+        expect(matIconDebugElement.textContent).toContain('notifications');
+    });
+
+    it('should return false when there are no unread notifications', () => {
+        component.unreadNotifications = [];
```

Comment:
Here we will get the notification object but with property with 'read': true
I think we need a notification object here with 'read': true property?

Reply:
added mock data for notifications (where 'read' is true) and also added an expect clause to check that unreadNotifications should be empty if all notifications are set to read:true.